### DISCUSSION TITLE
How to do "Message Flashing" like flask?

### DISCUSSION THREAD

**Discussion by 4n1qz5skwv** [litestar-org/litestar#1454] [General]
https://flask.palletsprojects.com/en/2.2.x/patterns/flashing/

```python
flash('Invalid password provided', 'error')
```

How to do this **LiteStar**?

**Comment by provinzkraut** [comment]
This would be considered an anti-pattern in Litestar, since it magically mutates hidden state that's being injected into the response data at some point.

The way to do this depends on what exactly it is you are trying to achieve. The example you provided should arguably have been a proper HTTP exception (`401 - Unauthorized`), so you would do `raise NotAuthorizedException("Invalid password provided")`.

**Reply by 4n1qz5skwv** [reply]
It's not about always error. some time it may be a simple piece of message. Every major library have this. This is a very helpful feature.

Django : https://docs.djangoproject.com/en/4.2/ref/contrib/messages/#adding-a-message
Flask: https://flask.palletsprojects.com/en/2.2.x/patterns/flashing/#flashing-with-categories
Quart: https://quart.palletsprojects.com/en/latest/reference/source/quart.helpers.html#quart.helpers.flash

**Reply by provinzkraut** [reply]
If all you want to do is add a message to consume in your templates, currently the best way is to pass them to the template context.

You can build an ad-hoc version of the flashing logic by using the request-local state, and then pass that into the context.

> Every major library have this

Well those are two: Django and Flask (and the async version of Flask). Most other major frameworks don't have this as far as I know; Sanic, Starlette, FastAPI certainly don't.

Our templating support is rather rudimentary since this framework is mostly focused on building APIs. But if you want to work on this you're more than welcome to add a PR. This could very well go into our contrib namespace.

**Reply by 4n1qz5skwv** [reply]
> pass them to the template context.

It's not that. It's something like a message is added to the session and on next page load those messages are retrieved from the session.


> Well those are two: **Django** and **Flask** 

😶


> Most other major frameworks don't have this

But they have extension: 
Sanic: https://github.com/lixxu/sanic-jinja2
Starlette (and the extend version of Starlette):  https://accent-starlette.github.io/starlette-core/messages/

> you're more than welcome to add a PR

All I'm asking is to consider it. StarLite (LiteStar) is going through a lot of changes , adding very exciting features. I just hope that at some point you consider this. That's All. 😇🤞👍

**Reply by provinzkraut** [reply]
> All I'm asking is to consider it. StarLite (LiteStar) is going through a lot of changes , adding very exciting features. I just hope that at some point you consider this. That's All. innocentcrossed_fingers+1

Well, I'm not saying no, but I also won't make any promises. You can however open an issue for this with an enhancement request, so it may be picked up by someone somewhere down the line!

**Reply by 4n1qz5skwv** [reply]
> open an issue for this

Thank you. Will do 😌

_(No option to convert this discussion into issue? Like, Issue to Discussion thing?)_

**Reply by provinzkraut** [reply]
> No option to convert this discussion into issue

There should be? Normally there's a button for that somewhere on this page :grimacing:

**Reply by 4n1qz5skwv** [reply]
> There should be? Normally there's a button for that somewhere on this page 😬

Done. Please add "enhancement" label.

https://github.com/litestar-org/litestar/issues/1455

**Comment by 4n1qz5skwv** [comment]
![abc](https://user-images.githubusercontent.com/76768193/230426996-8dbdc06c-62ae-4181-ab0c-8a955b20be89.PNG)

_**(not real... for now)**_
